### ISSUE TITLE
[BSI][20210511][v2.2] iOS Insufficient Escaping in vCard Construction

### ISSUE THREAD

**Issue by BSI-TF-CWA** [corona-warn-app/cwa-app-ios#2672]
Rating: Low

During the creation of the vCard file format, the iOS app is insufficiently escaping the semicolon character. The semicolon
acts as a separator between values of a multi-value property and must be escaped when it occurs within such a
value.
The iOS app replaces each occurrence of the semicolon character (;) with the compound value \; where the backslash
acts as an escape character. However, the app fails to escape the backslash character itself. This character needs to
be escaped whenever it occurs in a value.
Therefore, the value \; will be escaped as \\;. Due to the escaping rules this will be treated as an escaped backslash
character (\\) with a subsequent, unescaped, semicolon character ;.
In the context of the application itself, this does not have any security implications. However, it allows the user to break
out of the context that was originally intended. It might therefore violate the integrity and structure of the vCard data.
Should the code curently used for escaping be reused in another part of the ecosystem, security problems may arise
depending of the context it is used in.

**Proof of Concept**
The code snippet below shows the missing replacement of the backslash character. Taken from src/xcode/ENA/ENA/Source/Scenes/AntigenTestProfile/Cells/QRCodeCellViewModel.swift:80
```
// removed line breaks
var output = input.filter { !"\n\r".contains($0) }
output = output.replacingOccurrences(of: ";", with: "\\;")
output = output.replacingOccurrences(of: ",", with: "\\,")
return output
```

---
Internal Tracking ID: [EXPOSUREAPP-7205](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-7205)

**Comment by dsarkar** [comment]
FYI PR https://github.com/corona-warn-app/cwa-app-ios/pull/2731

**Comment by Ein-Tim** [comment]
@BSI-TF-CWA 

Version 2.3 has been released to the App Store yesterday, this version should fix this.
Can you confirm the fix?

**Comment by heinezen** [comment]
@Ein-Tim Full fix will come with CWA release 2.4.

----
Corona-Warn-App Open Source Team

**Comment by Ein-Tim** [comment]
PR #2930, merging into  branch `release/2.5.x`.

**Comment by dsarkar** [comment]
Quote:

> BSI Pentest confirms the issue has been successfully fixed in 2.5
